homeassistant/components/unifi/controller.py

```diff
@@ -312,19 +308,30 @@ def _async_check_for_stale(self, *_: datetime) -> None:
         for unique_id in unique_ids_to_remove:
             del self._heartbeat_time[unique_id]
 
-    async def async_update_device_registry(self) -> None:
-        """Update device registry."""
-        if self.mac is None:
-            return
+    @property
+    def device_info(self) -> DeviceInfo:
+        """UniFi controller device info."""
+        assert self.config_entry.unique_id is not None
 
-        device_registry = dr.async_get(self.hass)
+        version: str | None = None
+        if sysinfo := next(iter(self.api.system_information.values()), None):
+            version = sysinfo.version
+
+        return DeviceInfo(
+            entry_type=DeviceEntryType.SERVICE,
+            identifiers={(UNIFI_DOMAIN, self.config_entry.unique_id)},
+            manufacturer=ATTR_MANUFACTURER,
+            model="UniFi Network Application",
+            name="UniFi Network",
+            sw_version=version,
+        )
 
-        device_registry.async_get_or_create(
-            config_entry_id=self.config_entry.entry_id,
-            connections={(CONNECTION_NETWORK_MAC, self.mac)},
```

Comment:
Did you intent to remove the mac ?

Reply:
Yes. As it's primary a software service hosted on a machine. Might use via_device later on to connect the service with the host hardware.